### PR TITLE
EDM-168 Parametrizable templates

### PR DESCRIPTION
**Pull request by celdrake**
In the Create Fleet / Edit fleet wizards, it's possible to use template variables in the following configuration items:
- Inline configs
- Git configs (all fields except targetRevision which will be implemented by the backend separately)

We'll show a banner letting the user know about this. 
Fleetless devices cannot use template variables.

I didn't include additional validations as the fields that support parameters don't have a well defined regexp of what's allowed or not.

template-variables

### CODE REVIEW

File: libs/ui-components/src/components/Device/EditDeviceWizard/steps/DeviceTemplateStep.tsx [hunk 1]
```diff
@@ -1,28 +1,37 @@
...
+import ConfigTemplateForm from './ConfigTemplateForm';
 
 export const deviceTemplateStepId = 'device-template';
 
 export const isDeviceTemplateStepValid = (errors: FormikErrors<DeviceSpecConfigFormValues>) => {
   return !errors.osImage && !errors.configTemplates;
 };
 
-const DeviceTemplateStep = () => {
+const templateCode = 'device.metadata.labels[key]';
```
Comment (avishayt) [inline libs/ui-components/src/components/Device/EditDeviceWizard/steps/DeviceTemplateStep.tsx]:
```suggestion
const templateCode = '{{ device.metadata.labels[key] }}';
```

Comment (celdrake) [reply libs/ui-components/src/components/Device/EditDeviceWizard/steps/DeviceTemplateStep.tsx]: Good catch, I removed the brackets unvoluntirely.

File: libs/ui-components/src/components/Device/EditDeviceWizard/steps/DeviceTemplateStep.tsx [hunk 2]
```diff
@@ -1,28 +1,37 @@
...
 
   return (
     <Grid span={8}>
       <Form>
+        {isFleet && (
+          <Alert isInline variant="info" title={t('Using template variables')}>
+            <Trans t={t}>
+              Add a variable using <strong>{templateCode}</strong> and it will be applied based on the devices labels.
+              Template variables can only be used in Inline and Git configurations.
+            </Trans>{' '}
```
Comment (rawagner) [inline libs/ui-components/src/components/Device/EditDeviceWizard/steps/DeviceTemplateStep.tsx]: Looks like this should not be here
```suggestion
            </Trans>
```

Comment (celdrake) [reply libs/ui-components/src/components/Device/EditDeviceWizard/steps/DeviceTemplateStep.tsx]: right, removed

**Review by rawagner — APPROVED** [review]

**Pull request merged**